tests/models/user.py

```diff
@@ -0,0 +1,221 @@
+        type=SubmissionTypes.SUGG_ACCEPT,
+        suggestion=suggestion,
+    )
+    sub.save()
+    unit = suggestion.unit
+    unit.target = suggestion.target
+    unit.submitted_by = suggestion.user
+    unit.save()
+
+
+def _test_user_merged(unit, src_user, target_user):
+    # TODO: test reviews and comments
+
+    if src_user.id:
+        assert src_user.submitted.count() == 0
+        assert src_user.suggestions.count() == 0
+
+    assert target_user.submitted.first() == unit
+    assert target_user.suggestions.first() == unit.get_suggestions().first()
+
+
+def _test_before_evil_user_updated(store, member, debug=False):
+
+    # Unit target was updated.
+    assert store.units.first().target_f == "Hello, world UPDATED"
+    assert store.units.first().submitted_by == member
+    assert store.units.first().commented_by == member
+
+    # But member also added a suggestion to the unit.
+    assert store.units.first().get_suggestions().count() == 1
+    assert store.units.first().get_suggestions().first().user == member
+
+    # And added a comment on the unit.
+    assert store.units.first().translator_comment == "NICE COMMENT"
+
+    # Only 1 unit round here.
+    assert store.units.count() == 1
+
+
+def _test_after_evil_user_updated(store, evil_member):
+
+    # Evil member has accepted their own suggestion.
+    assert store.units.first().target_f == "EVIL SUGGESTION"
+    assert store.units.first().submitted_by == evil_member
+
+    # And rejected member's.
+    assert store.units.first().get_suggestions().count() == 0
+
+    # And added their own comment.
+    assert store.units.first().commented_by == evil_member
+    assert store.units.first().translator_comment == "EVIL COMMENT"
+
+    # Evil member has added another unit.
+    assert store.units.count() == 2
+    assert store.units[1].target_f == "Goodbye, world EVIL"
+    assert store.units[1].submitted_by == evil_member
+
+
+def _test_revision_increased(store, last_revision):
+    current_revision = store.get_max_unit_revision()
+    assert current_revision > last_revision
+    return current_revision
+
+
+def _test_user_purging(store, member, evil_member, purge):
+    assert store.units.first().target_f == ""
+    last_revision = store.get_max_unit_revision()
+
+    # Member updates first unit, adding a suggestion.
+    _create_submission_and_suggestion(store, member)
+    _create_comment_on_unit(store.units.first(), member, "NICE COMMENT")
+
+    # Test state before evil user has updated.
+    _test_before_evil_user_updated(store, member, evil_member)
+
+    # Revision has increased
+    last_revision = _test_revision_increased(store, last_revision)
+
+    # evil_member makes following changes:
+    #   - rejects member's suggestion on unit
+    #   - changes unit
+    #   - adds another suggestion on unit
+    #   - accepts their own suggestion
+    #   - adds a comment on unit
+    #   - adds another unit
+    _reject_suggestion(store.units.first().get_suggestions().first(),
```

Comment:
Thats a fair point - i think i was trying to establish how the review process worked when i copied out that code